app/models/katello/repository.rb

```diff
@@ -680,6 +680,49 @@ def docker_meta_tag_count
       DockerMetaTag.in_repositories(self.id).count
     end
 
+    # a master repository actually has content (rpms, errata, etc) in the pulp repository.  For these repositories, we can use the YumDistributor
+    # to generate metadata and can index the content from pulp, or for the case of content view archives without filters, can also use the YumCloneDistributor
+    #
+    def master?
+      !self.yum? || # non-yum repos
+          self.in_default_view? || # default content view repos
+          (!self.in_default_view? && self.archive? && !self.content_view.composite) || # non-composite content view archive repos
+          (self.content_view.composite? && self.component_source_repositories.count > 1) # composite archive repo with more than 1 source repository
+    end
+
+    # a link repository has no content in the pulp repository and serves as a shell.  It will always be empty.  Only the YumCloneDistributor can be used
+    # to publish yum metadata, and it cannot be indexed from pulp, but must have its indexed associations copied from another repository.
+    def link?
+      !master?
+    end
+
+    def target_repository
+      fail _("This is not a linked repository") if master?
+      return nil if self.archived_instance.nil?
+
+      if self.environment_id? && self.archived_instance.master?
+        self.archived_instance
+      elsif self.environment_id
+        self.archived_instance.target_repository || self.archived_instance #try to get the target_repository but fallback to archived_instance if nil
+      else #must be a composite archive repo, with only one component repo
```

Comment:
I am guessing this 'else' gets called  from the previous ```self.archived_instance.target_repository``` call

Reply:
correct!